### ISSUE TITLE
Unnecessary complexity in repo.sls states re permissions

### ISSUE THREAD

**Issue by nixjdm** [terminal-labs/rambo#122]
Both the hg and git cloning states perform the clone as root, and then have a separate state setting normal permissions. I presume we can just clone it as a right user and not need the second state.

https://github.com/terminal-labs/rambo/blob/master/rambo/salt_resources/states/hg/repo.sls#L5-L14
https://github.com/terminal-labs/rambo/blob/master/rambo/salt_resources/states/git/repo.sls#L5-L14

**Comment by nixjdm** [comment]
@alan-cugler A good first salt issue

**Comment by nixjdm** [comment]
This is not as easy as I thought. The complexity is that the deescalated user needs the appropriate keys to clone from a remote server. The root user has those keys, but the deescalated user doesn't. I'm making a new ticket for that though. #188 should be done before this ticket.
